cores/esp8266/uart.cpp

```diff
@@ -892,7 +892,7 @@ inline void
 uart_write_char_delay(const int uart_nr, char c)
 {
     while(uart_tx_fifo_full(uart_nr))
-        delay(0);
+        yield();
```

Comment:
Looking at `uart_write_char_delay`, it is used by the `uart{0,1}_write_char` and those specific write funcs are installed as `ets_install_putc1(...)` when debug mode is set. Does this mean that SYS calls this part, not CONT?

Reply:
This is right, `esp_schedule()` should be also called here.
So `uart{n}_write_char()`  should be in IRAM right ?
And `uart_write_char_delay()` is inline, maybe the attribute `__attribute__((always_inline))` should be added ?

Reply:
re IRAM and inline. It seems to be inlined anyway, no symbol is created for any functions down the chain besides delay(). Is it a concern for sdk code in any way?
Other q i think is answered below.

Reply:
@mcspr 
`inline` is an indication (like `register` which is deprecated in latest gcc version I believe) and compiler is free to not honor this request, which is fair.
But in that case, the function needs to be in iram because it can be called from an ISR.
Furthermore we can't declare a function to be at the same time inline and in iram.
So the proposal above is to force the compiler to always use this function inline.